src/azul/time.py

```diff
@@ -106,3 +112,45 @@ def parse_http_date(http_date: str, base_time: Optional[float] = None) -> float:
         return http_date.timestamp()
     else:
         return base_time + float(http_date)
+
+
+dcp2_datetime_format = '%Y-%m-%dT%H:%M:%S.%f%z'
+
+
+def to_dcp2_datetime_string(d: datetime) -> str:
+    """
+    Convert a tz-aware (UTC) datetime into a '2020-01-01T00:00:00.000000Z'
+    formatted string.
+
+    >>> to_dcp2_datetime_string(datetime(2020, 1, 1, tzinfo=timezone.utc))
+    '2020-01-01T00:00:00.000000Z'
+
+    >>> to_dcp2_datetime_string(datetime(2020, 1, 1))
+    Traceback (most recent call last):
+    ...
+    azul.RequirementError: Not a UTC timezone datetime.datetime(2020, 1, 1, 0, 0)
+    """
+    # `d.tzinfo` could have one of two possible 'UTC' values depending where the
+    # datetime value came from: 'pytz.utc' if the value was fetched from a
+    # BigQuery TIMESTAMP column, or 'timezone.utc' if the 'datetime' was
+    # converted from a string using 'strptime'
+    require(d.tzinfo in (pytz.utc, timezone.utc), f'Not a UTC timezone {d!r}')
```

Comment:
There must be a better way to check for UTC other than enumerating all possible tzinfo instances that could represent UTC.

Please don't use format strings in asserts and requires. Please create an additional entry in the C-guide to formalize this. I am tired of repeating myself.